### PR TITLE
Await websocket connection to be established in push and pull transfers

### PR DESCRIPTION
**Pull request by Bassel17**
### What does it do?

- Moved connectToWebsocket to provider utils so it can be used by the providers
- await until websocket connection is formed and handle potential errors gracefully

### Why is it needed?

- To handle errors better and throw accurate errors

### How to test it?

- Everything should work just fine for both push and pull

### CODE REVIEW

**Review by innerdvations — COMMENTED** [review]
LGTM just want to see if anyone has any comment on my comment

File: packages/core/data-transfer/src/strapi/providers/utils.ts [hunk 1]
```diff
@@ -119,6 +119,29 @@ export const createDispatcher = (ws: WebSocket) => {
   };
 };
 
+type WebsocketParams = ConstructorParameters<typeof WebSocket>;
+type Address = WebsocketParams[0];
+type Options = WebsocketParams[2];
```
[the remark this answers is not in the document]
Comment (Bassel17) [reply packages/core/data-transfer/src/strapi/providers/utils.ts:124]: I don't have a preference, it just looked cleaner

**Review by innerdvations — APPROVED** [review]
LGTM

**Review by christiancp100 — APPROVED** [review]
LGTM 🍝

**Pull request merged**